compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2254,26 +2278,40 @@ class Typer extends Namer
     trees mapconserve (typed(_))
 
   def typedStats(stats: List[untpd.Tree], exprOwner: Symbol)(implicit ctx: Context): List[Tree] = {
+    val (stats1, _) = typedStatsAndGetFacts(stats, exprOwner)
+    stats1
+  }
+
+  def typedStatsAndGetFacts(stats: List[untpd.Tree], exprOwner: Symbol)(implicit ctx: Context): (List[Tree], FlowFacts) = {
     val buf = new mutable.ListBuffer[Tree]
     val enumContexts = new mutable.HashMap[Symbol, Context]
       // A map from `enum` symbols to the contexts enclosing their definitions
-    @tailrec def traverse(stats: List[untpd.Tree])(implicit ctx: Context): List[Tree] = stats match {
+    @tailrec def traverse(stats: List[untpd.Tree], facts: FlowFacts)(implicit ctx: Context): (List[Tree], FlowFacts) = stats match {
       case (imp: untpd.Import) :: rest =>
         val imp1 = typed(imp)
         buf += imp1
-        traverse(rest)(ctx.importContext(imp, imp1.symbol))
+        traverse(rest, facts)(ctx.importContext(imp, imp1.symbol))
       case (mdef: untpd.DefTree) :: rest =>
         mdef.removeAttachment(ExpandedTree) match {
           case Some(xtree) =>
-            traverse(xtree :: rest)
+            traverse(xtree :: rest, facts)
           case none =>
-            typed(mdef) match {
+            import untpd.modsDeco
+            val ctx1 = if (ctx.explicitNulls) {
+              mdef match {
+                case mdef: untpd.ValDef if ctx.owner.is(Method) && !mdef.mods.is(Lazy | Implicit) =>
```

Comment:
I do not see why `Implicit` should be handled specially.

Reply:
We use the scheme you proposed, which doesn't require handling Implicit specially.